compiler.ts

```diff
@@ -95,10 +96,18 @@ export function augmentEnv(
     newGlobals.set(v.name, Number(globalAddr));
     mm.addGlobal(globalAddr);
   });
-  // for rg
-  const rgAddr = mm.staticAlloc(4n);
-  newGlobals.set("rg", Number(rgAddr));
-  mm.addGlobal(rgAddr);
+  // encoding for var rngi
+  for (let index = lastCount + 1; index <= forCount; index++) {
+    let rgAddr = mm.staticAlloc(4n);
+    newGlobals.set("rng" + index, Number(rgAddr));
+    mm.addGlobal(rgAddr);
+  }
+  // set bars for indexing in for loops
+  for (let index = lastCount + 1; index <= forCount; index++) {
+    let idxAddr = mm.staticAlloc(4n);
+    newGlobals.set("idx" + index, Number(idxAddr));
+    mm.addGlobal(idxAddr);
+  }
```

Comment:
Are these used as pointers in any capacity? Otherwise, they should not be marked as roots. From one of my tests, I'm guessing that they are being used as untagged numbers.

Reply:
What do you mean by pointers here? I don't quite understand. These are variables that save the information of `for loop` objects

Reply:
Are they ever used to store pointers to values? Or are they only used as numbers (if they are, are they tagged as primitives?)

As in, does the GC need to scan them to check object reachability?

Reply:
Oh yeah, the first one, `rgAddr` are pointers to `for loop` objects, and the `idxAddr` are numbers.

Reply:
@Quincy0v0 Sorry, lack of sleep made me misread that. Just remove `mm.addGlobal(idxAddr)`